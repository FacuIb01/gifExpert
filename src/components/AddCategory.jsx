import { useState } from "react";

export const AddCategory = ({ onNewCategorie }) => {
	const [inputValue, setInputValue] = useState("");

	const onInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const onSubmit = (event) => {
		event.preventDefault();

		if (inputValue.trim().length <= 1) return;

		onNewCategorie(inputValue.trim());
		setInputValue("");
	};

	return (
		<form aria-label="form"
			onSubmit={(event) => {
				onSubmit(event);
			}}
		>
			<input
				type="text"
				placeholder="Buscar gifs.."
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};
