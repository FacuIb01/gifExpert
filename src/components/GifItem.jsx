import { useState } from "react";

export const GifItem = ({ url, title }) => {
	const [Copiado, setCopiado] = useState(false);

	const handleCopyClick = () => {
		navigator.clipboard.writeText(url);
		setCopiado(true);
	};

	return (
		<div className="card">
			<img src={url} alt={title} />
			<p>
				{title}

				<button onClick={handleCopyClick}>{Copiado ? "âœ…" : "ðŸ“‹"}</button>
			</p>
		</div>
	);
};
