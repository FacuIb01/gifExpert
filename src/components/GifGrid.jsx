import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category, onDeleteCategorie }) => {
	const { images, isLoading } = useFetchGifs(category);

	const onDelete = () => {
		onDeleteCategorie(category);
	};

	return (
		<>
			<h3>
				{category} <button onClick={onDelete}>Borrar Categoria</button>
			</h3>
			{isLoading ? <h4>Cargando</h4> : null}
			<div className="card-grid">
				{images.map((image) => {
					return <GifItem key={image.id} {...image} />;
				})}
			</div>
		</>
	);
};
