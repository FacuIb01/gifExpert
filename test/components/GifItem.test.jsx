import {GifItem} from '../../src/components/GifItem'
import { render, screen } from '@testing-library/react'


describe('Test sobre el componente GifItem', () => { 

    const titulo = "Saitama"
    const url = "https://media0.giphy.com/media/v1.Y2lkPWMxNzAzYTRmdjV0bmpheTY5bnVmcTM1ZDR2MHd5eDMweHhhOTRya3lsbXEwbzhwYyZlcD12MV9naWZzX3NlYXJjaCZjdD1n/VXJWhaO7afRe/giphy.gif"
    
    test("Debe hacer match con el SnapShot", () => {
        const {container} = render(<GifItem url= 'asasasa' title="dale boca"/>)
        expect(container).toMatchSnapshot()
    })


    test("el url debe ser el pasado por parametro", () => {
        render(<GifItem url={url} title={titulo}/>)
        expect(screen.getByRole("img").src).toBe(url)
        expect(screen.getByRole("img").alt).toBe(titulo)
    })

    test("el texto del titulo debe existir", () => {
        render(<GifItem url={url} title={titulo}/>)
        expect(screen.findAllByText(titulo)).toBeTruthy()
    })
 })