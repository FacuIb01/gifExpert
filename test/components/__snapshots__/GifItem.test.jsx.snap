// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test sobre el componente GifItem Debe hacer match con el SnapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="dale boca"
      src="asasasa"
    />
    <p>
      dale boca
      <button>
        ðŸ“‹
      </button>
    </p>
  </div>
</div>
`;
